upsert/upsert.go

```diff
@@ -0,0 +1,45 @@
+package upsert
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+var (
+	//ErrInvalidResponse is returned when the Versioning API returns a non 200 status
+	ErrInvalidResponse = errors.New("invalid status returned by search.api.ch.gov.uk")
+)
+
+type Upsert interface {
+	// SendViaApi provides ability to post to search api
+	SendViaAPI(data string) (int, error)
+}
+
+// Template config contains recipient address, http client and send email endpoint
+type Template struct {
```

Comment:
This struct contains the implementations of the interface. Why isn't it called APIUpsert or HTTPUpsert to match the interface name?